app/lib/katello/repo_discovery.rb

```diff
@@ -172,14 +174,70 @@ def http_crawl(resume_point)
               @found << page_url.to_s
             else
               @to_follow << link.to_s if should_follow?(link.path)
-            end
-          end
+      end
+      end
           page.discard_doc! #saves memory, doc not needed
           []
         end
       end
     end
 
+    def spidr_proxy_details
+      details = {}
+
+      if proxy
+        details[:host] = proxy_host
+        details[:port] = proxy_port
+        details[:user] = proxy.username
+        details[:password] = proxy.password
+      end
+
+      details
+    end
+
+    def spidr_crawl(url)
+      user = @upstream_username if @upstream_username
+      password = @upstream_password if @upstream_password
+      #Spidr crawl the url over every url and add urls with repodata to found
+      Spidr.site(url, proxy: spidr_proxy_details) do |spider|
+        spider.authorized.add(url, user, password)
+        spider.every_url do |url|
+          if @found.include?(url.to_s.split('repodata/').first)
+            puts ">>> Skipping #{url.to_s}"
```

Comment:
These should probably be `Rails.logger.debug`s, no?

Reply:
Ya..I will clean this up and move it to ready for review.. 😻